src/librustc/mir/interpret/error.rs

```diff
@@ -565,8 +565,10 @@ impl fmt::Debug for UnsupportedOpInfo<'tcx> {
 pub enum ResourceExhaustionInfo {
     /// The stack grew too big.
     StackFrameLimitReached,
-    /// The program ran into an infinite loop.
-    InfiniteLoop,
+    /// The program ran for too long.
+    ///
+    /// The exact limit is set by the `const_eval_limit` attribute.
+    TimeLimitReached,
```

Comment:
Given that this is not about "time" but evaluation steps, maybe `StepLimitReached` would be better?